File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res=0;
        int n=s.size();
        for(int i=0;i<n;i++){
            int j=1;
            res++;
            while(i-j>=0 && i+j<n){
                if(s[i+j]==s[i-j]){;
                    j++;
                    res++;
                }
                else break;
            }
        }
        for(int i=0;i<n-1;i++){
            if(s[i]==s[i+1]){
                res++;
                int j=1;
                while(i-j>=0 && i+j+1<n){
                    if(s[i-j]==s[i+j+1]){
                        res++;
                        j++;
                    }
                    else break;
                }        

            }
        }
        return res;       
        
    }
};

File: Data Structures & Algorithms/minimum-window-with-characters/submission-0.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        if t=="":
            return ""
        
        countt,window = {},{}

        for c in t:
            countt[c] = 1+countt.get(c,0)

        have,need = 0,len(countt)
        res,reslen = [-1,-1], float("infinity")
        l=0
        for r in range(len(s)):
            c=s[r]
            window[c]=1+window.get(c,0)

            if c in countt and window[c]==countt[c]:
                have+=1

            while have==need:
                if (r-l+1) <reslen:
                    res=[l,r]
                    reslen = r-l+1
                window[s[l]]-=1

                if s[l] in countt and window[s[l]]==countt[s[l]]-1:
                    have-=1
                l+=1
        l,r =res
        return s[l:r+1] if reslen!=float("infinity") else ""
                